src/services/workflow.service.js

```diff
@@ -29,6 +31,70 @@ class LiveWorkflowService extends GQuery {
   }
 }
 
-const workflowService = new LiveWorkflowService()
+class SubscriptionWorkflowService extends GQuery {
+  constructor () {
+    super(/* enableWebSockets */ true)
+    /**
+     * @type {object}
+     */
+    this.observable = null
+  }
+
+  destructor () {
+    if (this.observable) {
+      this.observable.unsubscribe()
+    }
+    this.observable = null
+  }
+
+  subscribe (view, query) {
+    this.observable = this.request()
+    return super.subscribe(view, query)
+  }
+
+  recompute () {
+    this.destructor()
+    super.recompute()
+    // query has been recomputed, so subscribe to request again
+    this.observable = this.request()
+  }
+
+  request () {
+    /**
+     * Perform a REST GraphQL request for all subscriptions.
+     */
+    if (process.env.NODE_ENV !== 'production') {
+      console.debug('graphql request:', this.query)
+    }
+    if (!this.query) {
+      return null
+    }
+    const vm = this
+    return this.apolloClient.subscribe({
+      query: this.query,
+      fetchPolicy: 'no-cache'
+    }).subscribe({
+      next (response) {
```

Comment:
We override `request` here because we call `.subscribe`, which returns an `Observable`. The next `.subscribe` is - by coincidence - a method used to listen whenever the observable is updated.

Without overriding it, we would break the existing behaviour of the `GQuery` and, consequentially, break `LifeWorkflowService` with no necessity.

Reply:
Can probably remove the old `request` method, or potentially merge these classes since we shouldn't be using the old system for subscriptions after this.